dialogue-core/src/main/java/com/palantir/dialogue/core/BalancedNodeSelectionStrategyChannel.java

```diff
@@ -310,6 +311,10 @@ private static void registerGauges(
         }
     }
 
+    private static int countDistinct(long[] longs) {
+        return Arrays.stream(longs).boxed().collect(Collectors.toSet()).size();
```

Comment:
```suggestion
        return Arrays.stream(longs).distinct().count();
```

Alternatively:

```suggestion
    private static boolean containsDistinct(long[] longs) {
        if (longs.length < 2) return true;
        for (int i = 1; i < longs.length; i++) {
            if (longs[0] != longs[i]) return false;
        }
        return true;
    }
```

Reply:
🤦 yep definitely going for the simple distinct!